### ISSUE TITLE
File is compiled without RPM_OPT_FLAGS 何解？

### ISSUE THREAD

**Issue by GoogleCodeExporter** [memeco/osd-lyrics#188]
```
What's your Linux distro?
SLE_10~SLE_11_SP1,opensuse 11.1~11.4 in OBS(opensuse build service)

RPM 建包过程出现大量错误：

I: File is compiled without RPM_OPT_FLAGS
W: osd-lyrics no-rpm-opt-flags <cmdline>:CharDistribution.cpp, JpCntx.cpp, 
LangBulgarianModel.cpp, LangCyrillicModel.cpp, LangGreekModel.cpp, 
LangHebrewModel.cpp, LangHungarianModel.cpp, LangThaiModel.cpp, 
nsBig5Prober.cpp, nsCharSetProber.cpp, nsEUCJPProber.cpp, nsEUCKRProber.cpp, 
nsEUCTWProber.cpp, nsEscCharsetProber.cpp, nsEscSM.cpp, nsGB2312Prober.cpp, 
nsHebrewProber.cpp, nsLatin1Prober.cpp, nsMBCSGroupProber.cpp, nsMBCSSM.cpp, 
nsSBCSGroupProber.cpp, nsSBCharSetProber.cpp, nsSJISProber.cpp, 
nsUTF8Prober.cpp, nsUniversalDetector.cpp, src/entry/impl.cpp

I: Program is using uninitialized variables.
   Note the difference between "is used" and "may be used"
W: osd-lyrics uninitialized-variable ol_osd_window.c:1474

I: A function uses a 'return;' statement, but has actually a value
   to return, like an integer ('return 42;') or similar.
W: osd-lyrics voidreturn ol_lrc_fetch_qianqian.c:264
W: osd-lyrics voidreturn ol_osd_window.c:1161, 1175
W: osd-lyrics voidreturn ol_player_exaile03.c:105
W: osd-lyrics voidreturn ol_player_quodlibet.c:57
W: osd-lyrics voidreturn ol_player_rhythmbox.c:184

I: Program is using implicit definitions of special functions.
   these functions need to use their correct prototypes to allow
   the lightweight buffer overflow checking to work.
     - Implicit memory/string functions need #include <string.h>.
     - Implicit *printf functions need #include <stdio.h>.
     - Implicit *printf functions need #include <stdio.h>.
     - Implicit *read* functions need #include <unistd.h>.
     - Implicit *recv* functions need #include <sys/socket.h>.
E: osd-lyrics implicit-fortify-decl ol_debug.c:71
E: osd-lyrics implicit-fortify-decl ol_lrc.c:108
E: osd-lyrics implicit-fortify-decl ol_osd_module.c:125
E: osd-lyrics implicit-fortify-decl ol_osd_render.c:128
E: osd-lyrics implicit-fortify-decl ol_player_banshee.c:208
E: osd-lyrics implicit-fortify-decl ol_player_moc.c:195
E: osd-lyrics implicit-fortify-decl ol_singleton.c:59

I: Program returns random data in a function
E: osd-lyrics no-return-in-nonvoid-function ol_config.c:311, 297, 287, 277, 267
E: osd-lyrics no-return-in-nonvoid-function ol_lrc_fetch_ui.c:55, 30
E: osd-lyrics no-return-in-nonvoid-function ol_main.c:138
E: osd-lyrics no-return-in-nonvoid-function ol_osd_window.c:248
E: osd-lyrics no-return-in-nonvoid-function ol_player_exaile03.c:356
E: osd-lyrics no-return-in-nonvoid-function ol_player_rhythmbox.c:61
E: osd-lyrics no-return-in-nonvoid-function ol_search_dialog.c:113, 69

I: Program returns random data in a function
E: osd-lyrics no-return-in-nonvoid-function ol_config.c:311, 297, 287, 277, 267
E: osd-lyrics no-return-in-nonvoid-function ol_lrc_fetch_ui.c:55, 30
E: osd-lyrics no-return-in-nonvoid-function ol_main.c:138
E: osd-lyrics no-return-in-nonvoid-function ol_osd_window.c:248
E: osd-lyrics no-return-in-nonvoid-function ol_player_exaile03.c:356
E: osd-lyrics no-return-in-nonvoid-function ol_player_rhythmbox.c:61
E: osd-lyrics no-return-in-nonvoid-function ol_search_dialog.c:113, 69
[  290.880103] SysRq : Power Off
[  292.414338] Power down.



```

Original issue reported on code.google.com by `hillwood...@gmail.com` on 22 Apr 2011 at 2:48

Attachments:
- [rawlog](https://storage.googleapis.com/google-code-attachments/osd-lyrics/issue-188/comment-0/rawlog)

**Comment by GoogleCodeExporter** [comment]
```
使用什么版本编译的？试试git里的develop版本，应该修复了绝�
��部分问题的
```

Original comment by `tigerso...@gmail.com` on 22 Apr 2011 at 3:57

**Comment by GoogleCodeExporter** [comment]
```
svn版依然出现这个错误。
```

Original comment by `hillwood...@gmail.com` on 23 Apr 2011 at 11:59

**Comment by GoogleCodeExporter** [comment]
```
svn仓库已经废弃了，请参照wiki里的build使用git仓库
```

Original comment by `tigerso...@gmail.com` on 23 Apr 2011 at 12:05

**Comment by GoogleCodeExporter** [comment]
```
git版依然报错，请看详细日志。
```

Original comment by `hillwood...@gmail.com` on 23 Apr 2011 at 1:25

Attachments:
- [rawlog](https://storage.googleapis.com/google-code-attachments/osd-lyrics/issue-188/comment-4/rawlog)

**Comment by GoogleCodeExporter** [comment]
```
你用的应该是master分支，需要转移到develop分支：

git checkout --track origin/develop
```

Original comment by `tigerso...@gmail.com` on 23 Apr 2011 at 1:35

**Comment by GoogleCodeExporter** [comment]
```
在哪个目录运行这个命令？
hillwood@hillwood-opensuse:~/Desktop/osd-lyrics> git checkout --track 
origin/develop
fatal: git checkout: branch develop already exists

```

Original comment by `hillwood...@gmail.com` on 23 Apr 2011 at 2:03

**Comment by GoogleCodeExporter** [comment]
```
Try
1. git branch develop remotes/origin/develop
2. git checkout develop
```

Original comment by `ppc52...@gmail.com` on 28 Apr 2011 at 10:09

**Comment by GoogleCodeExporter** [comment]
```
编 rpm 时用的是怎样的 specfile ？
```

Original comment by `robinlee...@gmail.com` on 28 Apr 2011 at 2:33

**Comment by GoogleCodeExporter** [comment]
```
Name:           osd-lyrics
Version:        0.3.20101217
Release:        0
Summary:        Osd-lyrics is a third-party lyrics display program.
Group:          Applications/Multimedia
License:        GPLv3   
URL:            http://code.google.com/p/osd-lyrics/
Source0:        http://osd-lyrics.googlecode.com/files/%{name}-%{version}.tar.gz
#Patch0:        makefilebug-svn422.patch
BuildRoot:      %{_tmppath}/%{name}-%{version}-build
BuildRequires:  update-desktop-files
BuildRequires:  gtk2-devel, dbus-1-glib-devel, libcurl-devel, libglade2-devel, 
libnotify-devel,sqlite3-devel, gcc, gcc-c++, glibc-devel 
BuildRequires:  autoconf, automake,gettext-devel, intltool, libtool
Requires:       gtk2, dbus-1-glib , libcurl, libglade2, libnotify, sqlite3

%description
OSD 
Lyrics是受lrcdis的启发而开发的一款第三方歌词显示程序，它��
�Linux下的多款播放器提供类似Windows下QQ音乐的歌词显示功能��
�并能自动从网络上下载歌词。
OSD Lyrics is inspired by lrcdis. With OSD Lyrics, you can enjoy an OSD lyrics 
show with various media players in Linux.

%prep
%setup -q -n osd-lyrics
#%patch0 -p1
#%if 0%{?suse_version} > 1130
#sed -i 's/0\.17/0.18/' po/Makefile*
#%endif

%build
./autogen.sh
%configure --prefix=%{_prefix} \
           --libdir=%{_libdir} \
           --disable-mpd \
           --disable-xmms2
make %{?_smp_mflags}

%install
rm -rf $RPM_BUILD_ROOT
%makeinstall
%suse_update_desktop_file -r osd-lyrics Player AudioVideo

%clean
rm -rf $RPM_BUILD_ROOT

%post
touch --no-create %{_datadir}/icons/hicolor &>/dev/null || :

%postun
if [ $1 -eq 0 ] ; then
    touch --no-create %{_datadir}/icons/hicolor &>/dev/null || :
    gtk-update-icon-cache %{_datadir}/icons/hicolor &>/dev/null || :
fi

%posttrans
gtk-update-icon-cache %{_datadir}/icons/hicolor &>/dev/null || :

%files
%defattr(-,root,root,-)
%doc AUTHORS ChangeLog INSTALL README
%doc COPYING
%{_bindir}/osd-lyrics
%{_datadir}/applications/osd-lyrics.desktop
%{_datadir}/icons/hicolor/
%{_datadir}/osd-lyrics/
%{_datadir}/locale/


%changelog

```

Original comment by `hillwood...@gmail.com` on 28 Apr 2011 at 2:38

**Comment by GoogleCodeExporter** [comment]
```
我感觉是源代码编写不规范造成。为了保证编译好的程序能��
�大程度的正常运行，OBS的容错率是极低的。OBS会对源代码进�
��一些检查，有些不规范的代码即使gcc可以正常编译通过，但
是OBS仍然会报错退出，即使打包会成功，OBS也不会通过。
```

Original comment by `hillwood...@gmail.com` on 28 Apr 2011 at 2:43

**Comment by GoogleCodeExporter** [comment]
```
确定使用了git里develop分支的最新版？现在已经在编译时启用�
��最严格的检查，不应该会有各种隐式错误的
```

Original comment by `tigerso...@gmail.com` on 5 May 2011 at 6:50

**Comment by GoogleCodeExporter** [comment]
```
Thx 搞定。
```

Original comment by `hillwood...@gmail.com` on 6 May 2011 at 12:26

**Comment by GoogleCodeExporter** [comment]
Original comment by `tigerso...@gmail.com` on 28 May 2011 at 1:34
- Changed state: **WontFix**
